File: Back-End/Models/UserModel.cs
using APIBackend.DataModel;
using APIBackEnd.Data;

namespace APIBackEnd.Models
{
    public class UserModel
    {
        public UserModel() { }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string Branch { get; set; }
        public string Address { get; set; }
        public DateTime DateOnboard { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public int Age { get; set; }
        public bool? IsAdmin { get; set; }
        public int? DefaultWareHouseId { get; set; }
        public WareHouses DefaultWareHouse { get; set; }
        public List<UserWareHouseModel> ListUserWareHouse { get; set; }

    }
}

File: Front-End/angular/src/app/model/user.model.ts
import { UserWareHouse } from "./userWareHouse.model";
import { Warehouse } from "./warehouse.model";

export class Users{
    id: number;
    name?: string;
    username?: string;
    password?: string;
    branch?: string;
    address?: string;
    dateOnboard?: Date;
    age?: number;
    dateOfBirth?: Date;
    isAdmin?: boolean;
    listUserWareHouse?: UserWareHouse[];
    defaultWareHouseId?: number;
    defaultWareHouse?: Warehouse;
}